tuplex/core/include/physical/TransformTask.h

```diff
@@ -182,8 +182,19 @@ namespace tuplex {
         void sinkOutputToHashTable(HashTableFormat fmt, int64_t outputDataSetID);
         HashTableSink hashTableSink() const { return _htable; } // needs to be freed manually!
 
-        void setOutputLimit(size_t limit) { _outLimit = limit; resetOutputLimitCounter(); }
-        void setOutputSkip(size_t numRowsToSkip) { _outSkipRows = numRowsToSkip; }
+        void setOutputTopLimit(size_t limit) {
+            _outTopLimit = limit;
+        }
+
+        void setOutputBottomLimit(size_t limit) {
+            _outBottomLimit = limit;
+        }
+
+        // maxOrder of infinity means disregarding the bottomLimit short circuit
+        static void setMaxOrderAndResetLimits(size_t maxOrder = std::numeric_limits<size_t>::max()) {
```

Comment:
can you add doxygen string here? Also, is there a reason why this is a static function?

Reply:
Yes, it is static because it doesn't use any member value of the class, only modifying global static value.